packages/react-devtools-shared/src/hydration.js

```diff
@@ -383,7 +383,9 @@ export function hydrate(
 
     const value = parent[last];
 
-    if (value.type === 'infinity') {
+    if (!value) {
+      return;
+    } else if (value.type === 'infinity') {
       parent[last] = Infinity;
     } else if (value.type === 'nan') {
```

Comment:
Maybe we can have same issue for Infinty ?

Reply:
yeah, I miss the scenario which value is Infinity(Infinity is Truthy). Any suggestion for me ?  I think i did not understand your PR #19786 (I mean all logic detail) is the biggest obstacle to fix this bug. 
```js
- cleaned.forEach((path: Array<string | number>) => {
+ [...new Set(cleaned)].forEach((path: Array<string | number>) => {
```
Is these code ok? When we loop cleaned array, we filter the duplicate item and don't need check the **parent[last]**?

Reply:
We don't have duplicate keys here actually. In your example I think we have two symbols with the same name and as I have to change the symbols into a `string` so that we can do the `postmessage` that's why we have this issue.

Reply:
I didn't write any **Symbol** in my business code,  Just **class(decorators)** 、**mobx**  and **react**. I think store and store prototype has same params is the reason. (mobx internal reactivity machine logic)
Mobx will add **@observable** params to the prototype, so getAllEnumerableKeys will return [..., 'finished', 'finished'].
> Look the codesandbox console or chrome devtools console
[Reproduce Codesandbox link](https://codesandbox.io/s/simple-mobx-todolist-forked-ysi4p) 
<div style="display:flex;justify-content:center;width=600px;">
		<p><img src="https://user-images.githubusercontent.com/14012511/103552344-b7524980-4ee6-11eb-82dd-92b10bb6872e.png" width="600px" />
	    </p>
</div>

Reply:
Thank you. I can now reproduce the problem with a basic object.
[Reproduce issue]( https://codesandbox.io/s/simple-mobx-todolist-forked-jogl8?file=/index.js)
Maybe @bvaughn we should not include inherited keys or find a way to avoid duplicates ?
![image](https://user-images.githubusercontent.com/15034695/103555877-37f65300-4eb1-11eb-8935-7dcc1ff4c8fc.png)

Reply:
`p.p` in the above example only has one value (42) even though there are "duplicates". Our code (`getAllEnumerableKeys`) just needs to handle the fact that `Object.keys` and `Object.getOwnPropertyDescriptors` may overlap.

Seems like the easiest way would be to just use a `Set` internally?
```js
export function getAllEnumerableKeys(
  obj: Object,
): Array<string | number | Symbol> {
  const keys = new Set();
  let current = obj;
  while (current != null) {
    const currentKeys = [
      ...Object.keys(current),
      ...Object.getOwnPropertySymbols(current),
    ];
    const descriptors = Object.getOwnPropertyDescriptors(current);
    currentKeys.forEach(key => {
      // $FlowFixMe: key can be a Symbol https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor
      if (descriptors[key].enumerable) {
        keys.add(key);
      }
    });
    current = Object.getPrototypeOf(current);
  }
  return Array.from(keys);
}
```

Or maybe I'm overlooking something. 😄

Reply:
Or maybe better yet just change the return type?

```diff
diff --git a/packages/react-devtools-shared/src/utils.js b/packages/react-devtools-shared/src/utils.js
index 2c796c222..e9ba55f5a 100644
--- a/packages/react-devtools-shared/src/utils.js
+++ b/packages/react-devtools-shared/src/utils.js
@@ -67,8 +67,8 @@ export function alphaSortKeys(
 
 export function getAllEnumerableKeys(
   obj: Object,
-): Array<string | number | Symbol> {
-  const keys = [];
+): Set<string | number | Symbol> {
+  const keys = new Set();
   let current = obj;
   while (current != null) {
     const currentKeys = [
@@ -79,7 +79,7 @@ export function getAllEnumerableKeys(
     currentKeys.forEach(key => {
       // $FlowFixMe: key can be a Symbol https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor
       if (descriptors[key].enumerable) {
-        keys.push(key);
+        keys.add(key);
       }
     });
     current = Object.getPrototypeOf(current);
@@ -728,7 +728,7 @@ export function formatDataForPreview(
       return data.toString();
     case 'object':
       if (showFormattedValue) {
-        const keys = getAllEnumerableKeys(data).sort(alphaSortKeys);
+        const keys = Array.from(getAllEnumerableKeys(data)).sort(alphaSortKeys);
 
         let formatted = '';
         for (let i = 0; i < keys.length; i++) {
```

Since we don't need an Array in some cases.

Reply:
My previous commit is 
```js
 return [...new Set(keys)];
```
But it broken unit test [circle ci report](https://app.circleci.com/pipelines/github/facebook/react/8611/workflows/ac137a52-cab8-48a3-96e1-ded0ca7b7bf4/jobs/249856)... :cry:

Reply:
I ran tests locally with the diff I posted in [my comment above](https://github.com/facebook/react/pull/20534#discussion_r551425609) and they all pass.

Reply:
Ok, I rebase my PR with your code 😆 , and add unit test later. ( Now it's 02:16 in shanghai, china. I need to sleep. Thank u, bvaughn

Reply:
Please rest! 🙇 

Ping me once you've added a test and I'll review again.

Reply:
@bvaughn Hi, I had finished the unit test, please review, thank u very much ! 😆

Reply:
Thanks. It's back in my queue.

Reply:
If no problem in this pull request, I hope you can release a new small version (eg: 4.10.2) for react devtools , that's very helpful for my team. Thank you !  bvaughn